src/sidebar/util/version-data.js

```diff
@@ -0,0 +1,87 @@
+'use strict';
+
+/**
+ * An object representing version, browser and user data
+ *
+ * @typedef {Object} VersionData
+ * @property {string} timestamp - Current Date as string
+ * @property {string} url - Current document URL or 'N/A' if no available URL
+ * @property {string} fingerprint - document PDF fingerprint or 'N/A'
+ * @property {string} account - user id; with display name if available or 'N/A'
+ *                              if not logged in
+ * @property {string} userAgent - current browser user agent
+ * @property {string} version - current client app version
+ */
+
+/**
+ * An object representing user info
+ *
+ * @typedef {Object} UserInfo
+ * @property {string=} userid
+ * @property {string=} displayName
+ */
+
+/**
+ * An object representing document metadata
+ *
+ * @typedef {Object} DocMetadata
+ * @property {string=} documentFingerprint - optional PDF fingerprint for
+ *                     current document
+ */
+
+/**
+ * An object representing document info
+ *
+ * @typedef {Object} DocumentInfo
+ * @property {string=} url - current document URL
+ * @property {DocMetadata} metadata - document metadata
+ */
+
+class VersionData {
+  /**
+   * @param {UserInfo} userInfo
+   * @param {DocumentInfo} documentInfo
+   * @param {Object} window_ - test seam
```

Comment:
You can use the names of DOM classes as types, for example `Window` or `HTMLElement`, so this can be `@param {Window} window_`.